adapters/audienceNetwork/facebook.go

```diff
@@ -184,8 +184,17 @@ func (this *FacebookAdapter) modifyImp(out *openrtb.Imp) error {
 		}
 
 		if out.Banner.H == nil {
-			return &errortypes.BadInput{
-				Message: fmt.Sprintf("imp #%s: banner height required", out.ID),
+			for _, f := range out.Banner.Format {
+				if _, ok := supportedBannerHeights[f.H]; ok {
+					h := f.H
+					out.Banner.H = &h
+					break
+				}
+			}
+			if out.Banner.H == nil {
+				return &errortypes.BadInput{
+					Message: fmt.Sprintf("imp #%s: banner height required", out.ID),
+				}
 			}
 		}
 
```

Comment:
This is a nitpick but we can avoid one extra lookup operation is we use an else statement instead
```
162 func (this *FacebookAdapter) modifyImp(out *openrtb.Imp) error {
163     impType, ok := resolveImpType(out)
164 +-- 22 lines: if !ok {------------------------------------------
186         if out.Banner.H == nil {
187             for _, f := range out.Banner.Format {
188                 if _, ok := supportedBannerHeights[f.H]; ok {
189                     h := f.H
190                     out.Banner.H = &h
191                     break
192                 }
193             }
194             if out.Banner.H == nil {
195                 return &errortypes.BadInput{
196                     Message: fmt.Sprintf("imp #%s: banner height required", out.ID),
197                 }
198             }
199 -       }
200 -
201 -       if _, ok := supportedBannerHeights[*out.Banner.H]; !ok {
202 +       } else if _, ok := supportedBannerHeights[*out.Banner.H]; !ok {
203             return &errortypes.BadInput{
204                 Message: fmt.Sprintf("imp #%s: only banner heights 50 and 250 are supported", out.ID),
205             }
206         }
207
208         /* This will get overwritten post-serialization */
209         out.Banner.W = openrtb.Uint64Ptr(0)
210         out.Banner.Format = nil
211         break
212     }
213
214     return nil
215 }
adapters/audienceNetwork/facebook.go
```

Reply:
Oh, yes, that will save one hash table lookup under certain circumstances. :)